internal/registry/registry.go

```diff
@@ -30,20 +30,23 @@ import (
 const (
 	// providerKeyPath is where provider info is stored in to indexer repo.
 	providerKeyPath = "/registry/pinfo"
+	// extendedProvidersKeyPath is where extended provider info is stored in to indexer repo.
+	extendedProvidersKeyPath = "/registry/epinfo"
 )
 
 var log = logging.Logger("indexer/registry")
 
 // Registry stores information about discovered providers
 type Registry struct {
-	actions   chan func()
-	closed    chan struct{}
-	closeOnce sync.Once
-	closing   chan struct{}
-	dstore    datastore.Datastore
-	filterIPs bool
-	providers map[peer.ID]*ProviderInfo
-	sequences *sequences
+	actions           chan func()
+	closed            chan struct{}
+	closeOnce         sync.Once
+	closing           chan struct{}
+	dstore            datastore.Datastore
+	filterIPs         bool
+	providers         map[peer.ID]*ProviderInfo
+	extendedProviders map[peer.ID]*ExtendedProviders
```

Comment:
What is the reason for keeping  a separate `extendedProviders` map instead of adding a `**ExtendedProviders` to the `ProviderInfo` type?

Reply:
Done, see previous comment. Marking this one as resolved.

Reply:
Just one question - is there any specific reason why `ExtendedProviders` has to be a double pointer? Or is this a typo?

Reply:
Typo!!